lib/tensorflow/tensor.rb

```diff
@@ -171,22 +171,22 @@ def find_type(data)
   #
   def ruby_array_to_c(array, type)
     c_array = []
-    if type == Tensorflow::TF_INT64
-      c_array = Tensorflow::Long_long.new(array.length)
+    if type == Tensorflow::Internal::TF_INT64
+      c_array = Tensorflow::Internal::Long_long.new(array.length)
       array.each_with_index { |value, i| c_array[i] = value }
-    elsif type == Tensorflow::TF_INT32
-      c_array = Tensorflow::Int.new(array.length)
+    elsif type == Tensorflow::Internal::TF_INT32
+      c_array = Tensorflow::Internal::Int.new(array.length)
       array.each_with_index { |value, i| c_array[i] = value }
-    elsif type == Tensorflow::TF_STRING
-      c_array = Tensorflow::String_Vector.new
+    elsif type == Tensorflow::Internal::TF_STRING
+      c_array = Tensorflow::Internal::String_Vector.new
       array.each_with_index { |value, i| c_array[i] = value }
-      c_array = Tensorflow::string_array_from_string_vector(c_array)
-    elsif type == Tensorflow::TF_COMPLEX128
-      c_array = Tensorflow::Complex_Vector.new
+      c_array = Tensorflow::Internal::string_array_from_string_vector(c_array)
+    elsif type == Tensorflow::Internal::TF_COMPLEX128
+      c_array = Tensorflow::Internal::Complex_Vector.new
       array.each_with_index { |value, i| c_array[i] = value }
-      c_array = Tensorflow::complex_array_from_complex_vector(c_array)
+      c_array = Tensorflow::Internal::complex_array_from_complex_vector(c_array)
     else
-      c_array = Tensorflow::Double.new(array.length)
+      c_array = Tensorflow::Internal::Double.new(array.length)
```

Comment:
I guess 

```ruby
case type
when Tensorflow::Internal::TF_INT64
  c_array = Tensorflow::Internal::Long_long.new(array.length)
  ...
when Tensorflow::Internal::TF_INT32
  Tensorflow::Internal::Int.new(array.length)
  ...
when
...
... 
else
  Tensorflow::Internal::Double.new(array.length)
end
```
would be cleaner, but might be off-topic from what you're doing in this PR.